core/src/main/java/com/rim/IceField/Game.java

```diff
@@ -117,4 +118,184 @@ public static void Turn(PlayerBase player) throws Exception {
             }
         }
     }
+
+    //Will be continuously called in the game loop.
+    public void UserInteraction(ArrayList<String> text, PlayerBase player)
+    {
+        if(text.get(0).equals("move ")) {
+            if (text.get(1).equals("NORTH")) {
+
+                try {
+                    player.move("north", this.getMap());
+                    System.out.println("Action accepted!");
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            } else if (text.get(1).equals("SOUTH")) {
+                try {
+                    player.move("south", this.getMap());
+                    System.out.println("Action accepted!");
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            } else if (text.get(0).equals("LEFT")) {
+                try {
+                    player.move("west", this.getMap());
+                    System.out.println("Action accepted!");
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            } else if (text.get(0).equals("RIGHT")) {
+                try {
+                    player.move("east", this.getMap());
+                    System.out.println("Action accepted!");
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
+        else if(text.get(0).equals("use")) {
+            if (text.get(1).equals("food")) {
+                player.useItem("Food");
+                System.out.println("Action accepted!");
+            } else if (text.get(1).equals("shovel")) {
+                player.useItem("Shovel");
+                System.out.println("Action accepted!");
+            } else if (text.get(1).equals("diving suit")) {
+                player.useItem("Diving Suit");
+                System.out.println("Action accepted!");
+            }
+        }
+        else if(text.get(0).equals("pick"))
+        {
+            System.out.println("Would you like to pick " + player.currentIceberg.getItem() + "press 1 for yes, 2 for no");
+            Scanner input = new Scanner(System.in);
+            int y = input.nextInt();
+            if(y == 1) {
+                try {
+                    player.pickItem();
+                    System.out.println("Action accepted!");
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
+        else if(text.get(0).equals("apply skill"))
+        {
```

Comment:
here you could even use the same list of possible directions as with the move method